### PR TITLE
add OnResult  for #409 2022-11-23

### PR DESCRIPTION
**Pull request by hktalent**
I found during project integration that it is more convenient to use the results

### CODE REVIEW

File: pkg/server/dns_server.go [hunk 1]
```diff
@@ -28,6 +28,7 @@ type DNSServer struct {
 	server        *dns.Server
 	customRecords *customDNSRecords
 	TxtRecord     string // used for ACME verification
+
```
Comment (ravlio) [inline pkg/server/dns_server.go:31]: extra newline

**Review by ravlio — APPROVED** [review]

File: pkg/server/dns_server.go (hunk 1 above)
Comment (hktalent) [reply pkg/server/dns_server.go:31]: you can remove it

Comment (hktalent) [reply pkg/server/dns_server.go]: The redundant blank lines have been removed. Please merge the main branches. Thank you very much

Comment (ravlio) [reply pkg/server/dns_server.go]: I guess you should merge your pull request by yourself after it is approved

**Pull request merged**